delphi-checks/src/main/java/au/com/integradev/delphi/utils/StatementUtils.java

```diff
@@ -55,15 +57,18 @@ public static boolean isRoutineInvocation(
     }
 
     NameDeclaration declaration = ((NameReferenceNode) name).getLastName().getNameDeclaration();
-    List<ExpressionNode> arguments = extractArguments(expression.getChild(1));
+    List<ArgumentNode> arguments = extractArguments(expression.getChild(1));
     return declaration instanceof RoutineNameDeclaration
         && ((RoutineNameDeclaration) declaration).fullyQualifiedName().equals(fullyQualifiedName)
-        && argumentListPredicate.test(arguments);
+        && argumentListPredicate.test(
+            arguments.stream()
+                .map(ArgumentNode::getExpression)
+                .collect(Collectors.toUnmodifiableList()));
```

Comment:
Considering the name of `argumentListPredicate`, it could make more sense to change its type to a `Predicate<List<ArgumentNode>>` instead. It's only used in two places, and one of those places only checks its size.